src/main/java/bio/terra/service/filedata/FileService.java

```diff
@@ -158,29 +168,38 @@ FSItem lookupFSItem(String datasetId, String fileId, int depth) {
      * It is intended for user-facing calls (e.g. from RepositoryApiController), not internal calls that may require
      * an exclusively locked dataset to be returned (e.g. file deletion).
      */
-    FSItem lookupFSItemByPath(String datasetId, String path, int depth) {
+    FSItem lookupFSItemByPath(String datasetId, String path, int depth) throws InterruptedException {
         Dataset dataset = datasetService.retrieveAvailable(UUID.fromString(datasetId));
         return fileDao.retrieveByPath(dataset, path, depth, true);
     }
 
     // -- snapshot lookups --
     public FileModel lookupSnapshotFile(String snapshotId, String fileId, int depth) {
-        return fileModelFromFSItem(lookupSnapshotFSItem(snapshotId, fileId, depth));
+        try {
+            return fileModelFromFSItem(lookupSnapshotFSItem(snapshotId, fileId, depth));
+        } catch (InterruptedException ex) {
+            throw new FileSystemExecutionException("Unexpected interruption during file system processing", ex);
+        }
     }
 
     public FileModel lookupSnapshotPath(String snapshotId, String path, int depth) {
-        FSItem fsItem = lookupSnapshotFSItemByPath(snapshotId, path, depth);
+        FSItem fsItem = null;
+        try {
+            fsItem = lookupSnapshotFSItemByPath(snapshotId, path, depth);
+        } catch (InterruptedException ex) {
+            throw new FileSystemExecutionException("Unexpected interruption during file system processing", ex);
```

Comment:
I don't totally understand the impact of the changes in this PR. It appears that interrupted exception are still caught, but we're throwing FileSystemExecutionExceptions instead. What is the impact of throwing the FileSystem Exception instead of the interrupted exception?

Reply:
There are two cases:
1. Code run inside of a flight must never hide the interrupted exception. Otherwise a normal pod failover wood caused the flight to beaten marked failed.
2. Code serving the REST API should catch the interrupted exception and replace it with a meaningful exception. Otherwise the caller gets a useless error message.

Reply:
It might be useful to add this to the description.